src/amo/components/Search/index.js

```diff
@@ -71,3 +89,8 @@ export default class SearchPage extends React.Component {
     );
   }
 }
+
+export default compose(
+  safeAsyncConnect([{ promise: loadSearchResultsIfNeeded }]),
```

Comment:
Yeah porting Search to saga will be awesome but later is a good idea. It'll require a lot of UX tweaks 😄